test/lifecycle_test.dart

```diff
@@ -629,7 +716,9 @@ void sharedLifecycleTests<T extends react.Component>({
         react_dom.render(LifecycleTest(newProps), mountNode);
 
         List calls = [
-          matchCall('componentWillReceiveProps', args: [newPropsWithDefaults], props: initialPropsWithDefaults),
+          isComponent2
+              ? null
```

Comment:
Why isn't this being tested for `Component2`?

Reply:
I believe the thought process was that `componentWillReceiveProps` is no longer getting called because of getSnapshotBeforeUpdate, so just exclude it from the test. Is that logic sound or is there a test that would be good to put there in its place?

Reply:
I believe the thought process was that componentWillReceiveProps is no longer getting called because of getSnapshotBeforeUpdate, so just exclude it from the test. Is that logic sound or is there a test that would be good to put there in its place?

Reply:
`componentWillReceiveProps` was removed from `Component2`: https://github.com/cleandart/react-dart/pull/169/files#diff-ec47402eaffd4414fd19bddea36f05d0L96